### ISSUE TITLE
Corrupt xlsx after updating

### ISSUE THREAD

**Issue by hbostic** [guyonroche/exceljs#35]
Hello, great lib,  I have an issue with the following code whereby the updated file become corrupt.  The val you see in code is a string version of a bool so "true" or "false", I did hardcode a string as val ("hello"), just to see if it was the bool that caused the issue, but the file still got corrupted.  Can you please take a look and see what I may be doing wrong, thanks.  If I am able to recover the file, it has the updated value, but my app is no longer able to read the file for subsequent parsing

function updateExcelEntry(inputPage,sheet,colKeyIndex,colIndexToUpdate,rowKey,val){
    //console.log("in excel " + inputPage);
    var workbook = new Excel.Workbook();
    workbook.xlsx.readFile(inputPage).then(function() {
            // use workbook
        //console.log("getting sheet " + sheet);
        var worksheet = workbook.getWorksheet(sheet);
        //console.log("getting keyCol " + colKeyIndex);
        var keyCol = worksheet.getColumn(colKeyIndex);
        keyCol.key = "key";
        var col = worksheet.getColumn(colIndexToUpdate);
        col.key = "colToUpdate";

        worksheet.eachRow(function(row, rowNumber) {

            //console.log("checking key val " + row.getCell("key").value);
            //console.log("vs rowKey " + rowKey);
            if(rowNumber !== 1 && row.getCell("key").value === rowKey){
                //console.log("Row " + rowNumber + " = " + JSON.stringify(row.values));
                row.getCell("colToUpdate").value = val;

            }
        });

    }).then(function() {
        return workbook.xlsx.writeFile(inputPage)
    }).then(function() {
        console.log("workbook updated " + inputPage);
    });
}

**Comment by ezintz** [comment]
Yes, great work so far, and yes, the current version 0.2.3 is producing corrupt files - I can confirm that. Just read and write the same file.

```js
var Excel = require('exceljs');

var filename = './filename.xlsx'
var workbook = new Excel.Workbook();

workbook
  .xlsx
  .readFile(filename)
  .then(function() {
    var worksheet = workbook.getWorksheet('Sheet');

    worksheet.eachRow((row, rowNumber) => {
      console.log(row.values);
    });
  })
  .then(function() {
    return workbook.xlsx.writeFile(filename + '_modified.xlsx');
  })
  .then(function() {
    console.log('Done!');
  });
```

**Comment by ricmed205** [comment]
I'm getting corrupt files also v0.2.3

**Comment by issame** [comment]
I am also having this issue using the last version of exceljs

**Comment by IonicaBizau** [comment]
Same issue here, using the last version:

```js
"use strict";

const Excel = require("exceljs")
    , unstream = require("unstream")
    ;


// read from a file
var workbook = new Excel.Workbook();
//var workbook = new Excel.stream.xlsx.WorkbookWriter(options);
workbook.xlsx.readFile("input.xlsx").then(function(x) {
    let worksheet = workbook.getWorksheet("My Sheet");
    worksheet.addRow([3, 'Michael']).commit();
    return workbook.xlsx.writeFile("output.xlsx")

    //workbook.xlsx.write(unstream({}, function(data, cb) {
    //    console.log(data.toString("base64"));
    //    cb(null, data.toString('base64'));
    //}).pipe(require("fs").createWriteStream("foooo.xlsx")));
}).then(function () {
    console.log("Saved");
});
```

**Comment by IonicaBizau** [comment]
The `csv` output works, tho: `return workbook.csv.writeFile("output.csv")`

**Comment by guyonroche** [comment]
Could someone attach a sample input.xlsx file I could use to test this? I've tried the above code on a simple sheet and it works ok

**Comment by IonicaBizau** [comment]
@guyonroche Just any `xlsx` file. I remember I was creating it in LibreOffice from scratch and it was just not working.

**Comment by SergioDonati** [comment]
I have same issue. After read the latest comments i make a test.
Make a simple script that only read and write on different file.

First test:
1. Create new file with LibreOffice
2. run the script
3. file corrupted

Second test:
1. Create new file with Google Docs and download as xlsx
2. run the script
3. works!! 

So something is different in LibreOffice/OpenOffice files.

**Comment by stephengardner** [comment]
Nothing here worked as an intended solution for me.  Any updates?

**Comment by vodkatad** [comment]
The same phenomenon described by SergioDonati happens to me (version 0.7.1).

**Comment by hytromo** [comment]
I can confirm that this is happening for any libreoffice-generated xlsx I read and then write (even if no changes are made)

Libreoffice info:
```
Version: 5.4.2.2
Build ID: 1:5.4.2-0ubuntu0.17.10.1
CPU threads: 8; OS: Linux 4.13; UI render: default; VCL: kde4; 
Locale: en-US (en_US.UTF-8); Calc: group
```

**Comment by DStoyanoff** [comment]
Hi,
I have the same issue with the following simple code:
```
const workbook1 = new 
const sheet1 = workbook1.addWorksheet("Sheet1");
sheet1.columns = [
      { header: "FirstName", key: "firstname" },
      { header: "LastName", key: "lastname" },
      { header: "Other Name", key: "othername" }
];

await workbook1.xlsx.writeFile(`export-${new Date().getTime()}.xlsx`)
      .catch((e) => console.log(e));
````

![image](https://user-images.githubusercontent.com/5079334/40829898-8b7f73a4-658d-11e8-8aec-72ce1ec5318e.png)

Can you advise?

**Comment by mercmobily** [comment]
I can sadly confirm the problem. Wasted 6 hours on this...

**Comment by MusicDev33** [comment]
Experiencing this problem as well. If I open the file and just write it without changing it, nothing goes wrong, but if I change one cell, I get this error.

**Comment by mercmobily** [comment]
I solved the problem by using:

https://www.npmjs.com/package/xlsx-populate

All sorts of small formatting problems simply disappeared.
I cannot recommend it enough.

**Comment by MusicDev33** [comment]
@mercmobily Are you using this as a replacement for ExcelJS or are you using the data validation part of xlsx-populate?

**Comment by mercmobily** [comment]
I am using it to take an Excel sheet, use it as a template, and change some of the cells programmatically.

**Comment by papelias** [comment]
> I solved the problem by using:
> 
> https://www.npmjs.com/package/xlsx-populate
> 
> All sorts of small formatting problems simply disappeared.
> I cannot recommend it enough.

Why won't you recommend it. I'm just curious.

**Comment by mercmobily** [comment]
No, no, I wrote "I cannot recommend it enough". Meaning, I DO recommend it

**Comment by lhousaine** [comment]
Hi, I am still facing this same error, after reading an xlsx file with exceljs, then update one or more worksheets, then export the resulted file.

**Comment by ConstantinTuca** [comment]
> I solved the problem by using:
> 
> https://www.npmjs.com/package/xlsx-populate
> 
> All sorts of small formatting problems simply disappeared. I cannot recommend it enough.

I also had the same problem and **xlsx-populate** solved it. Thank you
